RandomErrorFixdebug.py

```diff
@@ -0,0 +1,233 @@
+            Pscore+=.25119
+        if ASCII is '(':
+            Pscore+=.19953
+        if ASCII is ')':
+            Pscore+=.15849
+        if ASCII is '*':
+            Pscore+=.12589
+        if ASCII is '+':
+            Pscore+=.10000
+        if ASCII is ',':
+            Pscore+=.07043
+        if ASCII is '-':
+            Pscore+=.06310
+        if ASCII is '.':
+            Pscore+=.05012
+        if ASCII is '/':
+            Pscore+=.03981
+        if ASCII is '0':
+            Pscore+=.03162
+        if ASCII is '1':
+            Pscore+=.02512
+        if ASCII is '2':
+            Pscore+=.01995
+        if ASCII is '3':
+            Pscore+=.01585
+        if ASCII is '4':
+            Pscore+=.01259
+        if ASCII is '5':
+            Pscore+=.01000
+        if ASCII is '6':
+            Pscore+=.00794
+        if ASCII is '7':
+            Pscore+=.00631
+        if ASCII is '8':
+            Pscore+=.00501
+        if ASCII is '9':
+            Pscore+=.00398
+        if ASCII is ':':
+            Pscore+=.00316
+        if ASCII is ';':
+            Pscore+=.00251
+        if ASCII is '<':
+            Pscore+=.00200
+        if ASCII is '=':
+            Pscore+=.00158
+        if ASCII is '>':
+            Pscore+=.00126
+        if ASCII is '?':
+            Pscore+=.00100
+        if ASCII is '@':
+            Pscore+=.00079
+        if ASCII is 'A':
+            Pscore+=.00063
+        if ASCII is 'B':
+            Pscore+=.00050
+        if ASCII is 'C':
+            Pscore+=.00040
+        if ASCII is 'D':
+            Pscore+=.00032
+        if ASCII is 'E':
+            Pscore+=.00025
+        if ASCII is 'F':
+            Pscore+=.00020
+        if ASCII is 'G':
+            Pscore+=.00016
+        if ASCII is 'H':
+            Pscore+=.00013
+        if ASCII is 'I':
+            Pscore+=.00010
+        if ASCII is 'J':
+            Pscore+=.00008
+    Pscorelist.append(Pscore)
+    return Pscorelist
+
+def phredscore(FastQ):
+    """reads lines in a file and searches for the lines before a phred score line and then applies phredvalues_version8()
+    :param fastq_fh: the file name/handle
+    :return: a list of all the phred score values for a file
+    """
+    Pscorelist=[]
+    for line in FastQ:
+        if '+' in line:
+            line=next(FastQ)
+            Pscorelist=phredvalues_version8(line)
+        elif '-' in line:
+            line=next(FastQ)
+            Pscorelist=phredvalues_version8(line)
+    return Pscorelist
+
+
+
+
+
+
+
+def errorproducer(FastQ):
+    """
+    seeds in errors to base pairs according to a total phred score for the corresponding phred score line
+    :param fastq_fh: the file handle/name
+    :return:lines with errors seeded into them
+    """
+
+    errorcount=0
+    totaltotalbase=0
+    Pscoreplacemark=0
+    dankhold=0
+    Pscorelist=[]
+    for line in FastQ:
+        print line
+        if dankhold==0:
+            Pscorelist=phredscore(FastQ)
```

Comment:
Here you are looping through all the lines in the fastq, so the outer loop should only execute once. This is because files are used in a streaming fashion, meaning you only read in one line at a time, once you read it in though, you can't go back unless you do a seek() call, which is not reccomended. Instead, to just read a single line, you can use FastQ.readline() to get back just one line, instead of looping through then in the phredscore routine.